### DISCUSSION TITLE
Next Major PHP SDK Version

### DISCUSSION THREAD

**Discussion by saralvasquez** [braintree/braintree_php#332] [Ideas]
## 👋  Welcome
We're using GitHub Discussions as a place to engage with the members of our community, to share ideas, and to support one another.

## 💡  Discussion Topic: Next Major Version of the Braintree PHP SDK
The Braintree SDK team is deciding what to include in our next major version release (v7). We are looking for some insight on:
- Language-specific improvements that you would like to see
- Features you'd like us to consider
- Pain points you'd like to see resolutions to

## 🗣 Let Us Know
Please use this thread to voice your opinions, concerns, or thoughts! We'd love to know how we can make the next version better for you.

**Comment by IncredibleHat** [comment]
One thing that would be great, is a configuration variable to have the SDK return error code/texts as part of the method call return object, instead of throwing exceptions all over the place and disrupting normal flow.

For example, what is the point of throwing an exception for something so simple as "customer ID not found"? That is **not** an error condition... its simply a result of zero.

Every use of the SDK we have to wrap the action with a monstrosity ??
```
	try {
		$response = $gateway->something()->action( $params );
	} catch(Braintree\Exception\Authentication $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\Authorization $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\Configuration $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\InvalidChallenge $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\InvalidSignature $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\ServerError $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\TooManyRequests $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\SSLCertificate $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\Timeout $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\GatewayTimeout $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\RequestTimeout $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\ServiceUnavailable $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\UpgradeRequired $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\NotFound $e) {
		// handle $e->getCode() / $e->getMessage()
	} catch(Braintree\Exception\Unexpected $e) {
		// handle $e->getCode() / $e->getMessage()
	}
```
